vms/avm/utxo/spender.go

```diff
@@ -29,18 +30,19 @@ type Spender interface {
 	// Arguments:
 	// - [utxos] contains assets ID and amount to be spend for each assestID
 	// - [kc] are the owners of the funds
-	// - [amounts] is the amount of funds that are available to be spent for each assetID
+	// - [toSpend] is the amount of funds that are available to be spent for each assetID
 	// Returns:
-	// - [amountsSpent] the amount of funds that are spent
 	// - [inputs] the inputs that should be consumed to fund the outputs
+	// - [outputs] the outputs produced
 	// - [signers] the proof of ownership of the funds being moved
 	Spend(
 		utxos []*avax.UTXO,
 		kc *secp256k1fx.Keychain,
-		amounts map[ids.ID]uint64,
+		toSpend map[ids.ID]uint64,
+		changeAddr ids.ShortID,
 	) (
-		map[ids.ID]uint64, // amountsSpent
 		[]*avax.TransferableInput, // inputs
+		[]*avax.TransferableOutput, // outputs
```

Comment:
On of the two main changes here: we don't return spent amount but we let Spend create the outputs as well.
The algorithm is the same in wallet builder.spend, with minimal changes.